drivers/spi/spi_context.h

```diff
@@ -134,11 +133,18 @@ static inline void spi_context_complete(struct spi_context *ctx, int status)
 		k_sem_give(&ctx->sync);
 	} else {
 		if (ctx->signal) {
+#ifdef CONFIG_SPI_SLAVE
```

Comment:
What about adding `&& defined(COUNT_RECV_FRAMES)` here and in all places where `recv_frames` is used?
Then, by defining `COUNT_RECV_FRAMES` right before including "spi_context.h" one could enable this feature when it is actually needed.

Reply:
And you won't get the async result working then. 

I understand your point about using existing register/variable for that, but is this really a relevant optimization? it's only a 32bits attributes with one increment operation. (And on a shim driver that needs to split up sg buffers I don't think it will make a lot of difference ;) )

Reply:
Though I think it wouldn't break async result, I will not insist. This was just a proposal. And you're the boss here, you decide. :)

Reply:
I agree completely. Let's get SPI closed finally.
And when a necessity of saving 4 bytes in RAM and a couple more in flash arises, I will know where to look for those. ;-)